rag-openai-qdrant/rag-openai-qdrant.ipynb

```diff
@@ -0,0 +1,672 @@
+       " 'ab94cc63ba53460fbad003a5e2b69e43']"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "client.add(\n",
+    "    collection_name=\"knowledge-base\",\n",
+    "    documents=[\n",
+    "        \"Qdrant is a vector database & vector similarity search engine. It deploys as an API service providing search for the nearest high-dimensional vectors. With Qdrant, embeddings or neural network encoders can be turned into full-fledged applications for matching, searching, recommending, and much more!\",\n",
+    "        \"Docker helps developers build, share, and run applications anywhere — without tedious environment configuration or management.\",\n",
+    "        \"PyTorch is a machine learning framework based on the Torch library, used for applications such as computer vision and natural language processing.\",\n",
+    "        \"MySQL is an open-source relational database management system (RDBMS). A relational database organizes data into one or more data tables in which data may be related to each other; these relations help structure the data. SQL is a language that programmers use to create, modify and extract data from the relational database, as well as control user access to the database.\",\n",
+    "        \"NGINX is a free, open-source, high-performance HTTP server and reverse proxy, as well as an IMAP/POP3 proxy server. NGINX is known for its high performance, stability, rich feature set, simple configuration, and low resource consumption.\",\n",
+    "        \"FastAPI is a modern, fast (high-performance), web framework for building APIs with Python 3.7+ based on standard Python type hints.\",\n",
+    "        \"SentenceTransformers is a Python framework for state-of-the-art sentence, text and image embeddings. You can use this framework to compute sentence / text embeddings for more than 100 languages. These embeddings can then be compared e.g. with cosine-similarity to find sentences with a similar meaning. This can be useful for semantic textual similar, semantic search, or paraphrase mining.\",\n",
+    "        \"The cron command-line utility is a job scheduler on Unix-like operating systems. Users who set up and maintain software environments use cron to schedule jobs (commands or shell scripts), also known as cron jobs, to run periodically at fixed times, dates, or intervals.\",\n",
+    "    ]\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b36bddd6",
+   "metadata": {},
+   "source": [
+    "## Retrieval Augmented Generation\n",
+    "\n",
+    "RAG changes the way we interact with Large Language Models. We're converting a knowledge-oriented task, in which the model may create a counterfactual answer, into a language-oriented task. The latter expects the model to extract meaningful information and generate an answer. That's what LLMs are supposed to be doing right.\n",
```

Comment:
Actually, LLMs are trained to predict the next token in a sequence, nothing more. So, all their "knowledge" is just a side effect of remembering the statistical relationships between tokens. In that sense, we should not rely on that "knowledge", but use their ability to extract and transform input data into a different format.

I liked the corrected version better since they only work with the natural language. There is no other interface.